src/utils/UnitSpherical/cap.jl

```diff
@@ -0,0 +1,66 @@
+#=
+# Spherical caps
+=#
+# Spherical cap implementation
+struct SphericalCap{T}
+    point::UnitSphericalPoint{T}
+    radius::T
+end
+
+SphericalCap(point::UnitSphericalPoint{T}, radius::Number) where T = SphericalCap{T}(point, convert(T, radius))
+SphericalCap(point, radius::Number) = SphericalCap(GI.trait(point), point, radius)
+function SphericalCap(::GI.PointTrait, point, radius::Number)
+    return SphericalCap(UnitSphereFromGeographic()(point), radius)
+end
+
+SphericalCap(geom) = SphericalCap(GI.trait(geom), geom)
+SphericalCap(t::GI.PointTrait, geom) = SphericalCap(t, geom, 0)
+# TODO: add implementations for line string and polygon traits
+# TODO: add implementations to merge two spherical caps
+# TODO: add implementations for multitraits based on this
+
+# TODO: this returns an approximately antipodal point...
+
+
+# TODO: exact-predicate intersection
+# This is all inexact and thus subject to floating point error
+function _intersects(x::SphericalCap, y::SphericalCap)
+    spherical_distance(x.point, y.point) <= max(x.radius, y.radius)
```

Comment:
Maybe I am completely misunderstanding, but shouldn't this be `spherical_distance(x.point, y.point) <= x.radius + y.radius`?

Reply:
Ah yes, good point - must have been thinking of something else when I wrote that 😅